search/searcher/search_term_prefix.go

```diff
@@ -55,6 +62,8 @@ func NewTermPrefixSearcher(indexReader index.IndexReader, prefix string, field s
 		return nil, err
 	}
 
+	qsearchers = nil
```

Comment:
I don't think this does anything useful (for which I could be wrong) and it's a style that we don't follow anywhere else, so I'd prefer we not do it.

My understanding is that `searcher` now has it's own []search.Searcher, (pointer/len/cap copied).  Assigning `qsearchers=nil` means this local variable no longer keeps the backing data live (but it is kept live by the new pointer inside the disjunction searcher.  Further, qsearchers goes out of scope at the end end of the function anyway, so we're not really accomplishing it significantly earlier either.